### ISSUE TITLE
Train SimCLR on 1 gpu Colab Fatal error -- HOOKS.Tensorboard not found 

### ISSUE THREAD

**Issue by duane-edgington** [facebookresearch/vissl#337]
## Instructions To Reproduce the 🐛 Bug:

1. what changes you made (`git diff`) or what code you wrote
```
ran copy of Train SimCLR on 1-gpu with VISSL tutorial on Colab. Same behavior if I ran without making a copy.
```
2. what exact command you run:
https://colab.research.google.com/drive/1Rt3Plt3ph84i1A-eolLFafybwjrBFxYe?usp=sharing
3. what you observed (including __full logs__):
```
!python3 run_distributed_engines.py \
    hydra.verbose=true \
    config=quick_1gpu_resnet50_simclr \
    config.DATA.TRAIN.DATA_SOURCES=[synthetic] \
    config.CHECKPOINT.DIR="./checkpoints" \
    config.HOOKS.TENSORBOARD_SETUP.USE_TENSORBOARD=true

####### overrides: ['hydra.verbose=true', 'config=quick_1gpu_resnet50_simclr', 'config.DATA.TRAIN.DATA_SOURCES=[synthetic]', 'config.CHECKPOINT.DIR=./checkpoints', 'config.HOOKS.TENSORBOARD_SETUP.USE_TENSORBOARD=true', 'hydra.verbose=true']
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/hydra/_internal/config_loader_impl.py", line 513, in _apply_overrides_to_config
    OmegaConf.update(cfg, key, value, merge=True)
  File "/usr/local/lib/python3.7/dist-packages/omegaconf/omegaconf.py", line 590, in update
    root[key_] = {}
  File "/usr/local/lib/python3.7/dist-packages/omegaconf/dictconfig.py", line 259, in __setitem__
    key=key, value=value, type_override=ConfigKeyError, cause=e
  File "/usr/local/lib/python3.7/dist-packages/omegaconf/base.py", line 101, in _format_and_raise
    type_override=type_override,
  File "/usr/local/lib/python3.7/dist-packages/omegaconf/_utils.py", line 629, in format_and_raise
    _raise(ex, cause)
  File "/usr/local/lib/python3.7/dist-packages/omegaconf/_utils.py", line 610, in _raise
    raise ex  # set end OC_CAUSE=1 for full backtrace
omegaconf.errors.ConfigKeyError: Key 'HOOKS' is not in struct
	full_key: config.HOOKS
	reference_type=Any
	object_type=dict

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "run_distributed_engines.py", line 194, in <module>
    hydra_main(overrides=overrides)
  File "run_distributed_engines.py", line 172, in hydra_main
    cfg = compose("defaults", overrides=overrides)
  File "/usr/local/lib/python3.7/dist-packages/hydra/experimental/compose.py", line 37, in compose
    with_log_configuration=False,
  File "/usr/local/lib/python3.7/dist-packages/hydra/_internal/hydra.py", line 512, in compose_config
    from_shell=from_shell,
  File "/usr/local/lib/python3.7/dist-packages/hydra/_internal/config_loader_impl.py", line 156, in load_configuration
    from_shell=from_shell,
  File "/usr/local/lib/python3.7/dist-packages/hydra/_internal/config_loader_impl.py", line 277, in _load_configuration
    ConfigLoaderImpl._apply_overrides_to_config(config_overrides, cfg)
  File "/usr/local/lib/python3.7/dist-packages/hydra/_internal/config_loader_impl.py", line 518, in _apply_overrides_to_config
    ) from ex
hydra.errors.ConfigCompositionException: Could not override 'config.HOOKS.TENSORBOARD_SETUP.USE_TENSORBOARD'.
To append to your config use +config.HOOKS.TENSORBOARD_SETUP.USE_TENSORBOARD=true

```
4. please simplify the steps as much as possible so they do not require additional resources to
   run, such as a private dataset.

## Expected behavior:

If there are no obvious error in "what you observed" provided above,
please tell us the expected behavior.

## Environment:

Provide your environment information using the following command:
```
wget -nc -q https://github.com/facebookresearch/vissl/raw/master/vissl/utils/collect_env.py && python collect_env.py

-------------------  ---------------------------------------------------------------
sys.platform         linux
Python               3.7.10 (default, May  3 2021, 02:48:31) [GCC 7.5.0]
numpy                1.19.5
Pillow               7.1.2
vissl                0.1.5 @/usr/local/lib/python3.7/dist-packages/vissl
GPU available        True
GPU 0                Tesla T4
CUDA_HOME            /usr/local/cuda
torchvision          0.6.1+cu101 @/usr/local/lib/python3.7/dist-packages/torchvision
hydra                1.0.6 @/usr/local/lib/python3.7/dist-packages/hydra
classy_vision        0.6.0.dev @/usr/local/lib/python3.7/dist-packages/classy_vision
tensorboard          1.15.0
apex                 0.1 @/usr/local/lib/python3.7/dist-packages/apex
cv2                  4.1.2
PyTorch              1.8.1+cu102 @/usr/local/lib/python3.7/dist-packages/torch
PyTorch debug build  False
-------------------  ---------------------------------------------------------------
PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v1.7.0 (Git Hash 7aed236906b1f7a05c0917e5257a1af05e9ff683)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 10.2
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_70,code=sm_70
  - CuDNN 7.6.5
  - Magma 2.5.2
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=10.2, CUDNN_VERSION=7.6.5, CXX_COMPILER=/opt/rh/devtoolset-7/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.8.1, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, 

CPU info:
-------------------  ------------------------------
Architecture         x86_64
CPU op-mode(s)       32-bit, 64-bit
Byte Order           Little Endian
CPU(s)               2
On-line CPU(s) list  0,1
Thread(s) per core   2
Core(s) per socket   1
Socket(s)            1
NUMA node(s)         1
Vendor ID            GenuineIntel
CPU family           6
Model                79
Model name           Intel(R) Xeon(R) CPU @ 2.20GHz
Stepping             0
CPU MHz              2199.998
BogoMIPS             4399.99
Hypervisor vendor    KVM
Virtualization type  full
L1d cache            32K
L1i cache            32K
L2 cache             256K
L3 cache             56320K
NUMA node0 CPU(s)    0,1
-------------------  ------------------------------
```

## When to expect Triage

VISSL devs and contributors aim to triage issues asap however, as a general guideline, we ask users to expect triaging in 1-2 weeks.

**Comment by prigoyal** [comment]
Hi @duane-edgington , thank you for reaching out about this. It seems like the issue is coming from
1) the command you are using installs VISSL pip packages - which is from Feb'21 
2) the code is running the master commands where the `HOOKS.TENSORBOARD_SETUP` change was done.

To solve this: 
1) in the tutorial, can you install VISSL from source instead: https://github.com/facebookresearch/vissl/blob/master/INSTALL.md#Install-from-source-in-PIP-environment and then try the command. This way your VISSL version and the commands will be in sync

NOTE also that we will work on releasing the new VISSL pip packages so that everything is in sync.

**Comment by duane-edgington** [comment]
Thank you for the timely response. I will try your solution, and look forward to your new pip package releases as well.

**Comment by pranavsinghps1** [comment]
> Hi @duane-edgington , thank you for reaching out about this. It seems like the issue is coming from
> 
> 1. the command you are using installs VISSL pip packages - which is from Feb'21
> 2. the code is running the master commands where the `HOOKS.TENSORBOARD_SETUP` change was done.
> 
> To solve this:
> 
> 1. in the tutorial, can you install VISSL from source instead: https://github.com/facebookresearch/vissl/blob/master/INSTALL.md#Install-from-source-in-PIP-environment and then try the command. This way your VISSL version and the commands will be in sync
> 
> NOTE also that we will work on releasing the new VISSL pip packages so that everything is in sync.

Building from source indeed solves the problem.
I tried the approach with regards to this notebook as well as other tutorial notebooks too.
Let me know if an update has to be made, I will gladly get it done!

**Comment by duane-edgington** [comment]
Thank you for the update. I spent time yesterday and this morning trying to install VISSL from source using PIP in the notebook, but I was unsuccessful in solving the problem (i.e. the notebook would still not run, and would fail on import vissl. 

Can you send me the cells or complete notebook with the successful install of VISSL and other needed packages so the tutorial notebook will execute with current packages, and everything in sync? 

This is very frustrating for a newbie trying to learn the framework... 

thank you 

Duane Edgington 
[ ***@***.*** | ***@***.*** ] 
github duane-edgington

**Comment by iseessel** [comment]
Hi @duane-edgington, can you take a look at: https://github.com/facebookresearch/vissl/pull/347, to see if this fixes your issue?

**Comment by duane-edgington** [comment]
Thank you for the note. 

No, I still have the same kind of error. 

Cell 2 error 

Successfully installed torch-1.9.0 
WARNING: The following packages were previously imported in this runtime:
  [torch]
You must restart the runtime in order to use newly installed versions. 
upon restart, I get the following error in cell 3 

--------------------------------------------------------------------------- 
ImportError Traceback (most recent call last) 
[ https://localhost:8080/# | <ipython-input-1-1a7d52a9609f> ] in <module> () 1 import vissl 2 import tensorboard ----> 3 import apex 4 import torch 


5 frames 
[ https://localhost:8080/# | /usr/local/lib/python3.7/dist-packages/apex/amp/_amp_state.py ] in <module> () 12 import collections . abc as container_abcs 13 else : ---> 14 from torch . _six import container_abcs 15 16 
ImportError : cannot import name 'container_abcs' from 'torch._six' (/usr/local/lib/python3.7/dist-packages/torch/_six.py) 
--------------------------------------------------------------------------- NOTE: If your import is failing due to a missing package, you can
manually install dependencies using either !pip or !apt.

To view examples of installing some common dependencies, click the
"Open Examples" button below.

**Comment by pranavsinghps1** [comment]
https://github.com/facebookresearch/vissl/pull/347#issuecomment-870067084
User-verified Fix addressed in #347 
CC : @iseessel @prigoyal

**Comment by iseessel** [comment]
@duane-edgington @pranavsinghps1 We have updated the tutorial to be in working order: https://vissl.ai/tutorials/Train_SimCLR_on_1_gpu_V0_1_6 in our latest vissl release: https://github.com/facebookresearch/vissl/releases/tag/v0.1.6
